controllers/idler/idler_controller.go

```diff
@@ -144,8 +144,23 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 					}
 					podLogger.Info("Pod deleted")
 				}
+				var appName string
+				var appType string
+				owners := pod.ObjectMeta.OwnerReferences
+				for _, owner := range owners {
+					if owner.Controller != nil && (owner.Kind == "Deployment" || owner.Kind == "ReplicaSet" ||
+						owner.Kind == "StatefulSet" || owner.Kind == "DeploymentConfig" ||
+						owner.Kind == "ReplicationController") {
+						appName = owner.Name
+						appType = owner.Kind
+						break
+					} else {
+						appName = pod.Name
+						appType = ""
+					}
```

Comment:
Nothing problematic with your code , but looking at `scaleControllerToZero` seems like we already have some logic to recursively find the root controller . 

For example , nn the case we have this hierarchy `Deployment --> ReplicaSet --> Pod` , this logic will report the `ReplicaSet` which might be fine, not a big issue. But I was wondering if in this case we want the `Deployment` name instead. 

ReplicaSet might be: `host-operator-controller-manager-5d7756db97` while Deployment name is: `host-operator-controller-manager`.

This is already a great improvement, so feel free to leave it as is if it's too complicated to reuse some of that logic for the recursive search .

Reply:
my understanding was that if its a replica we are going to scale to 0, we want the customer to know that we reduced the replica and not the main deployment..?

Reply:
In case of Deployments, as @mfrancisc mentioned there is a ReplicaSet too. Which is just an implementation detail of the Deployment. This is why we always look for the top parent and scale that parent to zero. If we will tell the user that we scaled down to zero the ReplicaSet then it would be misleading. It's not what we scaled down. And what is more importantly if the user tries to scale up the ReplicaSet instead of the Deployment then that will be auto-reverted by the Deployment. The Deployment is the ultimate controller here.

I would extend our `scaleControllerToZero()` function so it also returns the type of the controller if any. We should also cover the standalone pod case (no controllers). Basically let's keep the type of the actual resource we scale down or delete and re-use it when creating a notification instead of re-calculating it again after we scaled it down/deleted.